README.md

```diff
@@ -1,5 +1,42 @@
-# cloud-thanos-store-discovery
+Mattermost Cloud Thanos Store Discovery
+====================================================
 
-This microservice is used to automatically register new Thanos Query targets with central Thanos deployment. 
+Mattermost Thanos store discovery tool is a microservice designed to work in a multi-cluster environment, with the purpose to automatically register new Thanos query endpoints in a central Thanos deployment.
 
-More details and information on usage and deployment coming soon.
+## How it works?
+
+This microservice needs to run in the same cluster and namespace that the central Thanos deployment is running. It can be deployed as a cronjob that runs every x amount of minutes and keeps the central Thanos Querier up to date with all the target endpoints. More details on how it works can be seen in the following diagram.
```

Comment:
```suggestion
This microservice needs to run in the same cluster and namespace where the central Thanos deployment is running. It can be deployed as a cronjob that runs every x amount of minutes and keeps the central Thanos Querier up to date with all the target endpoints. More details on how it works can be seen in the following diagram.
```